app/stac_api/validators_serializer.py

```diff
@@ -527,3 +529,33 @@ def validate_query_parameters_post_search(self, query_param):
             logger.error(
                 'Query contains the non-allowed parameter(s): %s', list(wrong_query_parameters)
             )
+
+
+def validate_uniqueness_and_create(model_class, validated_data):
```

Comment:
The idea of this method is nice and I guess there's no way around it, in order to properly treat the integrity error in the atomic context. It's just a bit "un-nice", that is validates and also creates in one step an inside the `validators_serializer.py`..

Reply:
@hansmannj  Exactly we cannot go without this function, however we could move it into serializer.py ?

Reply:
The world will keep on turning no matter if it stays here or is moved into `serializer.py` :smile: 
Just leave it here.